tests/foreman/api/test_docker.py

```diff
@@ -592,39 +502,34 @@ def test_add_docker_repo_to_composite_content_view(self, name):
         @Feature: Docker
 
         """
-        prod_id = entities.Product(
-            organization=self.org_id
-        ).create_json()['id']
-
-        repo_id = _create_repository(prod_id, name)['id']
+        product = entities.Product(organization=self.org).create()
+        repo = _create_repository(product, name)
 
         # Create content view and associate docker repo
         content_view = entities.ContentView(
-            organization=self.org_id, composite=False
-        ).create_json()
-        _add_repo_to_content_view(repo_id, content_view['id'])
-
-        new_attrs = entities.ContentView(id=content_view['id']).read_json()
-        self.assertIn(repo_id, new_attrs['repository_ids'])
-
-        # Publish it...
-        entities.ContentView(id=content_view['id']).publish()
-        # ... and grab its version ID (there should only be one version)
-        new_attrs = entities.ContentView(id=content_view['id']).read_json()
-        version_id = new_attrs['versions'][0]['id']
-
-        # Create composite content view and associate content view to
-        # it
-        comp_content_view_id = entities.ContentView(
-            organization=self.org_id, composite=True
-        ).create_json()['id']
-        _add_content_view_to_composite_view(
-            comp_content_view_id, version_id)
-
-        new_attrs = entities.ContentView(
-            id=comp_content_view_id
-        ).read_json()
-        self.assertIn(version_id, new_attrs['component_ids'])
+            composite=False,
+            organization=self.org,
+        ).create()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
+        self.assertIn(repo.id, [repo_.id for repo_ in content_view.repository])
+
+        # Publish itand grab its version ID (there should only be one version)
+        content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), 1)
+
+        # Create composite content view and associate content view to it
+        comp_content_view = entities.ContentView(
+            composite=True,
+            organization=self.org,
+        ).create()
+        comp_content_view.component = content_view.version
+        comp_content_view = comp_content_view.update(['component'])
+        self.assertIn(
+            content_view.version[0].id,
+            [component.id for component in comp_content_view.component]
```

Comment:
The `update()` method returns a fresh content view object, just like `create()` and `read()` do.